src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoProjectDecorator.java

```diff
@@ -220,14 +218,10 @@ public void logout() throws IOException {
                     found.get().setState(ComponentState.PUSHED);
                     found.get().setPath(path);
                 } else {
-                    try {
-                        ComponentKind kind = getComponentKind(path);
-                        ComponentInfo.Builder builder = new ComponentInfo.Builder();
-                        ComponentInfo info = builder.withComponentKind(kind).build();
-                        components.add(Component.of(comp.getName(), ComponentState.NOT_PUSHED, path, info));
-                    } catch (IOException e) {
-                       throw e;
-                    }
+                    ComponentKind kind = getComponentKind(path);
+                    ComponentInfo.Builder builder = new ComponentInfo.Builder();
+                    ComponentInfo info = builder.withComponentKind(kind).build();
+                    components.add(Component.of(comp.getName(), ComponentState.NOT_PUSHED, path, info));
```

Comment:
these are the only lines added, they're not catched any more. The rest are removals. Can you please elaborate how these removals fix things when using the newer kubernetes-client?

Reply:
this is an unnecessary try-catch clause. the method already throws IOE and catch clause only re-throw IOE.